services/motion/builtin/wrapperFrame.go

```diff
@@ -0,0 +1,159 @@
+package builtin
+
+import (
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	pb "go.viam.com/api/component/arm/v1"
+
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/utils"
+)
+
+// WrapperFrame is a frame which merges the planning and localization frames of a PTG base.
+// This struct is used so that we do not break abstractions made in CheckPlan.
+type wrapperFrame struct {
+	name              string
+	localizationFrame referenceframe.Frame
+	executionFrame    referenceframe.Frame
+	seedMap           map[string][]referenceframe.Input
+	fs                referenceframe.FrameSystem
+}
+
+func newWrapperFrame(
+	localizationFrame, executionFrame referenceframe.Frame,
+	seedMap map[string][]referenceframe.Input,
+	fs referenceframe.FrameSystem,
+) referenceframe.Frame {
+	return &wrapperFrame{
+		name:              executionFrame.Name(),
+		localizationFrame: localizationFrame,
+		executionFrame:    executionFrame,
+		seedMap:           seedMap,
+		fs:                fs,
+	}
+}
+
+// Name returns the name of the wrapper frame's execution frame's name.
+func (wf *wrapperFrame) Name() string {
+	return wf.name
+}
+
+// Transform returns the associated pose given a list of inputs.
+func (wf *wrapperFrame) Transform(inputs []referenceframe.Input) (spatialmath.Pose, error) {
+	if len(inputs) != len(wf.DoF()) {
+		return nil, referenceframe.NewIncorrectInputLengthError(len(inputs), len(wf.DoF()))
+	}
+	pf := referenceframe.NewPoseInFrame(wf.Name(), spatialmath.NewZeroPose())
+	tf, err := wf.fs.Transform(wf.sliceToMap(inputs), pf, referenceframe.World)
```

Comment:
May you please expand on this comment a little?
Here I am following how we do things in `solverFrame.go/Transform()`

Are you suggesting that I:
1. determine which subset of inputs relates to the execution and localization Frame
2. with these subsets transform the execution and localization Frame and get some result?, e.g. poseExection, and poseLocalization
3. compose(poseExecution, poselocalization)?
If so, this method does not make use of the framesystem we may use, and using the fs seems like a cleaner abstraction.

Reply:
That is indeed what I am suggesting. Doing this removes the need to have a fs or the seedMap at all.

`solverFrame` can operate on any sort of frames, in any arrangement; it does not have the luxury of any prior knowledge about the system it is operating on. Thus it must use a framesystem, which of course works, but is a more complex abstraction and another things to keep track of.

This is a private struct used within its package; we know *precisely* which frames it will operate on, and precisely how they are arranged. There's no need to deal with the potential for static frames in between, or other unrelated frames, etc.